xonsh/history.py

```diff
@@ -110,7 +110,8 @@ def files(self, only_unlocked=False):
         """
         _ = self  # this could be a function but is intimate to this class
         # pylint: disable=no-member
-        xdd = os.path.abspath(builtins.__xonsh_env__.get('XONSH_DATA_DIR'))
+        xdd = os.path.expanduser(builtins.__xonsh_env__.get('XONSH_DATA_DIR'))
+        xdd = os.path.abspath(xdd)
```

Comment:
@melund thanks for the assist! Appreciated.
This is now fixed in the latest (additional) push.